tempodb/blocklist/poller.go

```diff
@@ -360,3 +485,19 @@ func sumTotalBackendMetaMetrics(blockMeta []*backend.BlockMeta, compactedBlockMe
 		compactedBlockMetaTotalBytes:   sumTotalBytesCBM,
 	}
 }
+
+func uuidInBlocklist(tenantID string, id uuid.UUID, list backend.Blocklist) bool {
+	for _, i := range list.Metas(tenantID) {
```

Comment:
this is very inefficient. we should request the list.Metas and list.CompactedMetas once and reuse them when checking for what's already in the blocklist

Reply:
Good call, I'll make the adjustments to pass in the variables we've already captured above.